### PR TITLE
56 - Add use cases to get the current authenticated user and log out

### LINKED ISSUE #56: Add use case to get the current authenticated user given an auth cookie
## Overview of the Feature Request
Add a use case that allows obtaining the current authenticated user, given an auth cookie that comes from the JSF UI.

The related API endpoint is: `/api/v1/users/:me`

Being the first operation that requires authentication, we will focus on session cookie API auth (currently enabled in the backend using a feature flag). API token auth will not be supported in this scope, but the changes must be implemented in a way that can be extended to support that authentication mechanism as well in the future.

## What kind of user is the feature intended for?
Dataverse frontend developers

## What inspired the request?
- https://github.com/IQSS/dataverse-frontend/issues/53

## What existing behavior do you want changed?
N/A

## Any brand new behavior do you want to add to Dataverse Frontend?
New use case for retrieving the current authenticated user

## Any related open or closed issues to this feature request?

- https://github.com/IQSS/dataverse/issues/9531
- https://github.com/IQSS/dataverse-frontend/issues/53

### PR DESCRIPTION
**Pull request by GPortas**
## What this PR does / why we need it:

- #55 Needs to be merged first!

Adds the following use cases to the package:

- Get the current authenticated user
- Logout

Both operations work given an auth cookie that comes from the JSF UI. It is the only auth mechanism added for the moment to support the SPA MVP. There is a TODO comment, in the ApiRepository file, providing more information and next steps on this.

## Which issue(s) this PR closes:

- Closes https://github.com/IQSS/dataverse-client-javascript/issues/56

## Special notes for your reviewer:

This PR is part of an end-to-end flow that covers three PRs from three different repositories:
- https://github.com/IQSS/dataverse-frontend/pull/67
- https://github.com/IQSS/dataverse/pull/9533

## Suggestions on how to test this:

Test the end-to-end flow following the frontend repository PR instructions:
- https://github.com/IQSS/dataverse-frontend/pull/67

## Is there a release notes update needed for this change?:

N/A

## Additional documentation:

N/A

### CODE REVIEW

**Review by pdurbin — COMMENTED** [review]
@GPortas this PR seems to have changes from the recent refactoring? Maybe we need a fresh one?

**Comment by GPortas** [comment]
> 

I have refreshed the branch with the latest develop changes. Thanks! @pdurbin

File: test/unit/auth/Logout.test.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    await sut.execute();
+  });
+
+  test('should return error result on repository error', async () => {
+    const authRepositoryStub = <IAuthRepository>{};
+    const testWriteError = new WriteError();
+    authRepositoryStub.logout = sandbox.stub().throwsException(testWriteError);
+    const sut = new Logout(authRepositoryStub);
+
+    let actualError: WriteError = undefined;
```
Comment (pdurbin) [inline test/unit/auth/Logout.test.ts:26]: VS Code is complaining about this line:

Screen Shot 2023-04-26 at 9 06 10 AM

**Review by pdurbin — COMMENTED** [review]
Overall this looks fine. I'm interested in testing this but ideally all three pull requests would be near final before we do so. The one for Dataverse proper is still being discussed.

I did `npm run test` and everything passed including tests like "logout".

VS Code is complaining about a line in a test. Not a big deal but it would be nice to not see red:

Screen Shot 2023-04-26 at 9 11 48 AM

**Review by pdurbin — DISMISSED** [review]
Approved. I've been been testing this with the frontend and backend PRs.

**Comment by pdurbin** [comment]
Please note that until we address this issue...

- https://github.com/IQSS/dataverse-frontend/issues/77

... I am taking on faith that the js-dataverse code (this repo) is the same as the tarball that has been put into the frontend PR. For more discussion:

- https://github.com/IQSS/dataverse-frontend/pull/67#discussion_r1179015263
- https://iqss.slack.com/archives/C04MJ76A34Z/p1682594807564159

**Review by kcondon — APPROVED** [review]

**Pull request merged**